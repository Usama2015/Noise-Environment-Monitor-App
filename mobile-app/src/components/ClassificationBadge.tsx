/**
 * ClassificationBadge Component
 *
 * Displays the noise classification category with color-coded badge,
 * icon, and descriptive text.
 *
 * Features:
 * - Color-coded badge (green/yellow/red)
 * - Category icon (emoji)
 * - Category label
 * - Optional description
 * - Confidence indicator
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface ClassificationBadgeProps {
  /** Noise category */
  category: NoiseCategory;
  /** Hex color code for badge */
  color: string;
  /** Icon for category (emoji) */
  icon: string;
  /** Optional description of noise type */
  description?: string;
  /** Optional confidence score (0-1) */
  confidence?: number;
  /** Optional detailed category info */
  categoryInfo?: string;
}

/**
 * ClassificationBadge component
 *
 * @example
 * ```tsx
 * <ClassificationBadge
 *   category="Normal"
 *   color="#FFC107"
 *   icon="ðŸŸ¡"
 *   description="Voice / Music"
 *   confidence={0.85}
 * />
 * ```
 */
export const ClassificationBadge: React.FC<ClassificationBadgeProps> = ({
  category,
  color,
  icon,
  description,
  confidence,
  categoryInfo,
}) => {
  return (
    <View style={styles.container}>
      {/* Main badge */}
      <View style={[styles.badge, { backgroundColor: color }]}>
        <Text style={styles.icon}>{icon}</Text>
        <Text style={styles.categoryText}>{category}</Text>
      </View>

      {/* Noise type description */}
      {description && (
        <Text style={styles.description}>{description}</Text>
      )}

      {/* Confidence indicator */}
      {confidence !== undefined && (
        <View style={styles.confidenceContainer}>
          <Text style={styles.confidenceLabel}>Confidence:</Text>
          <View style={styles.confidenceBar}>
            <View
              style={[
                styles.confidenceFill,
                {
                  width: `${confidence * 100}%`,
                  backgroundColor: getConfidenceColor(confidence),
                },
              ]}
            />
          </View>
          <Text style={styles.confidenceValue}>
            {(confidence * 100).toFixed(0)}%
          </Text>
        </View>
      )}

      {/* Category information */}
      {categoryInfo && (
        <View style={styles.infoContainer}>
          <Text style={styles.infoText}>{categoryInfo}</Text>
        </View>
      )}
    </View>
  );
};

/**
 * Get color for confidence bar based on confidence level
 */
function getConfidenceColor(confidence: number): string {
  if (confidence >= 0.8) return '#4CAF50'; // High confidence - green
  if (confidence >= 0.6) return '#FFC107'; // Medium confidence - yellow
  return '#FF9800'; // Low confidence - orange
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginVertical: 16,
  },
  badge: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 20,
    paddingVertical: 12,
    borderRadius: 25,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.15,
    shadowRadius: 4,
    elevation: 3,
  },
  icon: {
    fontSize: 24,
    marginRight: 8,
  },
  categoryText: {
    fontSize: 20,
    fontWeight: 'bold',
    color: '#fff',
  },
  description: {
    fontSize: 16,
    color: '#666',
    marginTop: 12,
    fontStyle: 'italic',
  },
  confidenceContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: 12,
    width: '80%',
  },
  confidenceLabel: {
    fontSize: 12,
    color: '#999',
    marginRight: 8,
    width: 70,
  },
  confidenceBar: {
    flex: 1,
    height: 8,
    backgroundColor: '#E0E0E0',
    borderRadius: 4,
    overflow: 'hidden',
  },
  confidenceFill: {
    height: '100%',
    borderRadius: 4,
  },
  confidenceValue: {
    fontSize: 12,
    fontWeight: '600',
    color: '#666',
    marginLeft: 8,
    width: 40,
    textAlign: 'right',
  },
  infoContainer: {
    marginTop: 12,
    paddingHorizontal: 16,
    paddingVertical: 8,
    backgroundColor: '#F5F5F5',
    borderRadius: 8,
    maxWidth: '90%',
  },
  infoText: {
    fontSize: 12,
    color: '#666',
    textAlign: 'center',
    lineHeight: 16,
  },
});
