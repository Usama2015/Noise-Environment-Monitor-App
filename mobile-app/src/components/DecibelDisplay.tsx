/**
 * DecibelDisplay Component
 *
 * Displays the current decibel level with a large, animated number
 * and visual color indicator based on noise category.
 *
 * Features:
 * - Large, prominent decibel value
 * - Smooth number transitions
 * - Color-coded background based on category
 * - Responsive sizing
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React, { useEffect, useRef } from 'react';
import { Animated, StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface DecibelDisplayProps {
  /** Current decibel value */
  decibels: number;
  /** Noise category for color coding */
  category: NoiseCategory;
  /** Optional label */
  label?: string;
}

/**
 * Get background color based on noise category
 */
function getCategoryBackgroundColor(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return '#E8F5E9'; // Light green
    case 'Normal':
      return '#FFF9C4'; // Light yellow
    case 'Noisy':
      return '#FFEBEE'; // Light red
  }
}

/**
 * Get text color based on noise category
 */
function getCategoryTextColor(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return '#2E7D32'; // Dark green
    case 'Normal':
      return '#F57C00'; // Dark orange
    case 'Noisy':
      return '#C62828'; // Dark red
  }
}

/**
 * DecibelDisplay component
 *
 * @example
 * ```tsx
 * <DecibelDisplay
 *   decibels={65.4}
 *   category="Normal"
 *   label="Current Level"
 * />
 * ```
 */
export const DecibelDisplay: React.FC<DecibelDisplayProps> = ({
  decibels,
  category,
  label = 'Sound Level',
}) => {
  // Animated value for smooth transitions
  const fadeAnim = useRef(new Animated.Value(1)).current;
  const scaleAnim = useRef(new Animated.Value(1)).current;

  // Pulse animation when decibel value changes significantly
  useEffect(() => {
    // Pulse effect
    Animated.sequence([
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 1.05,
          duration: 100,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 0.8,
          duration: 100,
          useNativeDriver: true,
        }),
      ]),
      Animated.parallel([
        Animated.timing(scaleAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }),
        Animated.timing(fadeAnim, {
          toValue: 1,
          duration: 200,
          useNativeDriver: true,
        }),
      ]),
    ]).start();
  }, [decibels, category, fadeAnim, scaleAnim]);

  const backgroundColor = getCategoryBackgroundColor(category);
  const textColor = getCategoryTextColor(category);

  return (
    <View style={styles.container}>
      {label && <Text style={styles.label}>{label}</Text>}

      <Animated.View
        style={[
          styles.displayBox,
          {
            backgroundColor,
            opacity: fadeAnim,
            transform: [{ scale: scaleAnim }],
          },
        ]}
      >
        <Text style={[styles.decibelValue, { color: textColor }]}>
          {decibels.toFixed(1)}
        </Text>
        <Text style={[styles.unit, { color: textColor }]}>dB</Text>
      </Animated.View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    marginVertical: 20,
  },
  label: {
    fontSize: 14,
    fontWeight: '500',
    color: '#666',
    marginBottom: 8,
    textTransform: 'uppercase',
    letterSpacing: 1,
  },
  displayBox: {
    paddingHorizontal: 40,
    paddingVertical: 30,
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    minWidth: 200,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 8,
    elevation: 4,
  },
  decibelValue: {
    fontSize: 64,
    fontWeight: 'bold',
    lineHeight: 72,
  },
  unit: {
    fontSize: 24,
    fontWeight: '600',
    marginTop: 4,
  },
});
