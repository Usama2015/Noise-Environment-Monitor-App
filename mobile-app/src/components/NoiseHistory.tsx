/**
 * NoiseHistory Component
 *
 * Displays a list of recent noise readings with decibel values
 * and classifications.
 *
 * Features:
 * - Scrollable list of readings
 * - Color-coded entries
 * - Timestamp display
 * - Limit to last N readings
 *
 * @see PROJECT_PLAN.md - Step 1.7: Basic UI Implementation
 */

import React from 'react';
import { ScrollView, StyleSheet, Text, View } from 'react-native';
import type { NoiseCategory } from '../services/NoiseClassifier';

export interface NoiseReading {
  /** Decibel value */
  decibels: number;
  /** Classification category */
  category: NoiseCategory;
  /** Timestamp */
  timestamp: Date;
  /** Optional description */
  description?: string;
}

export interface NoiseHistoryProps {
  /** Array of noise readings */
  readings: NoiseReading[];
  /** Maximum number of readings to display */
  maxReadings?: number;
  /** Show timestamps */
  showTimestamps?: boolean;
}

/**
 * Get color dot for category
 */
function getCategoryDot(category: NoiseCategory): string {
  switch (category) {
    case 'Quiet':
      return 'ðŸŸ¢';
    case 'Normal':
      return 'ðŸŸ¡';
    case 'Noisy':
      return 'ðŸ”´';
  }
}

/**
 * Format timestamp for display
 */
function formatTimestamp(date: Date): string {
  const now = new Date();
  const diffMs = now.getTime() - date.getTime();
  const diffSecs = Math.floor(diffMs / 1000);

  if (diffSecs < 60) {
    return `${diffSecs}s ago`;
  }

  const diffMins = Math.floor(diffSecs / 60);
  if (diffMins < 60) {
    return `${diffMins}m ago`;
  }

  const diffHours = Math.floor(diffMins / 60);
  if (diffHours < 24) {
    return `${diffHours}h ago`;
  }

  return date.toLocaleTimeString();
}

/**
 * NoiseHistory component
 *
 * @example
 * ```tsx
 * <NoiseHistory
 *   readings={recentReadings}
 *   maxReadings={10}
 *   showTimestamps={true}
 * />
 * ```
 */
export const NoiseHistory: React.FC<NoiseHistoryProps> = ({
  readings,
  maxReadings = 10,
  showTimestamps = true,
}) => {
  // Limit to max readings and sort by timestamp (newest first)
  const displayReadings = readings
    .slice(-maxReadings)
    .reverse();

  if (displayReadings.length === 0) {
    return (
      <View style={styles.emptyContainer}>
        <Text style={styles.emptyText}>No readings yet</Text>
        <Text style={styles.emptySubtext}>
          Start monitoring to see history
        </Text>
      </View>
    );
  }

  return (
    <View style={styles.container}>
      <Text style={styles.title}>Recent Readings</Text>

      <ScrollView
        style={styles.scrollView}
        contentContainerStyle={styles.scrollContent}
        showsVerticalScrollIndicator={false}
      >
        {displayReadings.map((reading, index) => (
          <View
            key={`${reading.timestamp.getTime()}-${index}`}
            style={styles.readingItem}
          >
            <View style={styles.readingLeft}>
              <Text style={styles.dot}>{getCategoryDot(reading.category)}</Text>
              <View style={styles.readingInfo}>
                <View style={styles.readingMain}>
                  <Text style={styles.decibelText}>
                    {reading.decibels.toFixed(1)} dB
                  </Text>
                  <Text style={styles.categoryText}>{reading.category}</Text>
                </View>
                {reading.description && (
                  <Text style={styles.descriptionText} numberOfLines={1}>
                    {reading.description}
                  </Text>
                )}
              </View>
            </View>

            {showTimestamps && (
              <Text style={styles.timestamp}>
                {formatTimestamp(reading.timestamp)}
              </Text>
            )}
          </View>
        ))}
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 20,
  },
  title: {
    fontSize: 16,
    fontWeight: '600',
    color: '#333',
    marginBottom: 12,
    paddingHorizontal: 16,
  },
  scrollView: {
    flex: 1,
    maxHeight: 300,
  },
  scrollContent: {
    paddingHorizontal: 16,
  },
  readingItem: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingVertical: 12,
    paddingHorizontal: 12,
    backgroundColor: '#FAFAFA',
    borderRadius: 8,
    marginBottom: 8,
  },
  readingLeft: {
    flexDirection: 'row',
    alignItems: 'center',
    flex: 1,
  },
  dot: {
    fontSize: 16,
    marginRight: 12,
  },
  readingInfo: {
    flex: 1,
  },
  readingMain: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  decibelText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#333',
    marginRight: 8,
  },
  categoryText: {
    fontSize: 14,
    color: '#666',
  },
  descriptionText: {
    fontSize: 11,
    color: '#999',
    fontStyle: 'italic',
    marginTop: 2,
  },
  timestamp: {
    fontSize: 11,
    color: '#999',
    marginLeft: 8,
  },
  emptyContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 40,
  },
  emptyText: {
    fontSize: 16,
    fontWeight: '600',
    color: '#999',
    marginBottom: 4,
  },
  emptySubtext: {
    fontSize: 13,
    color: '#BBB',
  },
});
