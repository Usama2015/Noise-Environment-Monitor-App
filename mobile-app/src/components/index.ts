/**
 * Component exports
 *
 * Central export file for all UI components
 */

export { DecibelDisplay } from './DecibelDisplay';
export type { DecibelDisplayProps } from './DecibelDisplay';

export { ClassificationBadge } from './ClassificationBadge';
export type { ClassificationBadgeProps } from './ClassificationBadge';

export { NoiseHistory } from './NoiseHistory';
export type { NoiseHistoryProps, NoiseReading } from './NoiseHistory';
