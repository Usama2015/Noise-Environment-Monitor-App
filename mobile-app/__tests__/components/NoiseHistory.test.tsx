/**
 * NoiseHistory Component Tests
 *
 * Tests for the noise history list component including rendering,
 * empty states, and data display.
 */

import React from 'react';
import { render } from '@testing-library/react-native';
import { NoiseHistory, NoiseReading } from '../../src/components/NoiseHistory';

// Helper to create test readings
function createReading(
  decibels: number,
  category: 'Quiet' | 'Normal' | 'Noisy',
  minutesAgo: number = 0,
  description?: string
): NoiseReading {
  const timestamp = new Date();
  timestamp.setMinutes(timestamp.getMinutes() - minutesAgo);

  return {
    decibels,
    category,
    timestamp,
    description,
  };
}

describe('NoiseHistory', () => {
  describe('Empty state', () => {
    it('should show empty message when no readings', () => {
      const { getByText } = render(
        <NoiseHistory readings={[]} />
      );

      expect(getByText('No readings yet')).toBeTruthy();
      expect(getByText('Start monitoring to see history')).toBeTruthy();
    });

    it('should not show title when empty', () => {
      const { queryByText } = render(
        <NoiseHistory readings={[]} />
      );

      expect(queryByText('Recent Readings')).toBeNull();
    });
  });

  describe('Rendering readings', () => {
    it('should render single reading', () => {
      const readings = [createReading(65.4, 'Normal')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('Recent Readings')).toBeTruthy();
      expect(getByText('65.4 dB')).toBeTruthy();
      expect(getByText('Normal')).toBeTruthy();
    });

    it('should render multiple readings', () => {
      const readings = [
        createReading(45.2, 'Quiet'),
        createReading(68.7, 'Normal'),
        createReading(82.1, 'Noisy'),
      ];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('45.2 dB')).toBeTruthy();
      expect(getByText('68.7 dB')).toBeTruthy();
      expect(getByText('82.1 dB')).toBeTruthy();
    });

    it('should show descriptions when provided', () => {
      const readings = [
        createReading(65, 'Normal', 0, 'Voice / Music'),
      ];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('Voice / Music')).toBeTruthy();
    });

    it('should not show descriptions when not provided', () => {
      const readings = [
        createReading(65, 'Normal'),
      ];

      const { queryByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(queryByText('Voice / Music')).toBeNull();
    });
  });

  describe('Category indicators', () => {
    it('should show green dot for Quiet', () => {
      const readings = [createReading(40, 'Quiet')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('ðŸŸ¢')).toBeTruthy();
      expect(getByText('Quiet')).toBeTruthy();
    });

    it('should show yellow dot for Normal', () => {
      const readings = [createReading(60, 'Normal')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('ðŸŸ¡')).toBeTruthy();
      expect(getByText('Normal')).toBeTruthy();
    });

    it('should show red dot for Noisy', () => {
      const readings = [createReading(80, 'Noisy')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('ðŸ”´')).toBeTruthy();
      expect(getByText('Noisy')).toBeTruthy();
    });
  });

  describe('Max readings limit', () => {
    it('should limit to maxReadings', () => {
      const readings = Array.from({ length: 20 }, (_, i) =>
        createReading(50 + i, 'Normal')
      );

      const { queryByText } = render(
        <NoiseHistory readings={readings} maxReadings={5} />
      );

      // Should show last 5 (newest)
      expect(queryByText('69.0 dB')).toBeTruthy(); // reading 19
      expect(queryByText('68.0 dB')).toBeTruthy(); // reading 18
      expect(queryByText('67.0 dB')).toBeTruthy(); // reading 17
      expect(queryByText('66.0 dB')).toBeTruthy(); // reading 16
      expect(queryByText('65.0 dB')).toBeTruthy(); // reading 15

      // Should not show older ones
      expect(queryByText('50.0 dB')).toBeNull(); // reading 0
      expect(queryByText('51.0 dB')).toBeNull(); // reading 1
    });

    it('should default to 10 readings', () => {
      const readings = Array.from({ length: 15 }, (_, i) =>
        createReading(50 + i, 'Normal')
      );

      const { queryByText } = render(
        <NoiseHistory readings={readings} />
      );

      // Should show last 10
      expect(queryByText('64.0 dB')).toBeTruthy(); // reading 14
      expect(queryByText('55.0 dB')).toBeTruthy(); // reading 5

      // Should not show first 5
      expect(queryByText('50.0 dB')).toBeNull(); // reading 0
    });
  });

  describe('Timestamps', () => {
    it('should show timestamps by default', () => {
      const readings = [createReading(65, 'Normal', 5)];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('5m ago')).toBeTruthy();
    });

    it('should hide timestamps when showTimestamps=false', () => {
      const readings = [createReading(65, 'Normal', 5)];

      const { queryByText } = render(
        <NoiseHistory readings={readings} showTimestamps={false} />
      );

      expect(queryByText(/ago/)).toBeNull();
    });

    it('should format seconds correctly', () => {
      const reading = createReading(65, 'Normal');
      reading.timestamp = new Date(Date.now() - 30 * 1000); // 30 seconds ago

      const { getByText } = render(
        <NoiseHistory readings={[reading]} />
      );

      expect(getByText(/\d+s ago/)).toBeTruthy();
    });

    it('should format minutes correctly', () => {
      const reading = createReading(65, 'Normal', 15);

      const { getByText } = render(
        <NoiseHistory readings={[reading]} />
      );

      expect(getByText('15m ago')).toBeTruthy();
    });

    it('should format hours correctly', () => {
      const reading = createReading(65, 'Normal');
      reading.timestamp = new Date(Date.now() - 2 * 60 * 60 * 1000); // 2 hours ago

      const { getByText } = render(
        <NoiseHistory readings={[reading]} />
      );

      expect(getByText('2h ago')).toBeTruthy();
    });
  });

  describe('Decibel formatting', () => {
    it('should format to 1 decimal place', () => {
      const readings = [createReading(65.456, 'Normal')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('65.5 dB')).toBeTruthy();
    });

    it('should handle integer values', () => {
      const readings = [createReading(70, 'Noisy')];

      const { getByText } = render(
        <NoiseHistory readings={readings} />
      );

      expect(getByText('70.0 dB')).toBeTruthy();
    });
  });

  describe('Snapshot tests', () => {
    it('should match snapshot for empty state', () => {
      const { toJSON } = render(
        <NoiseHistory readings={[]} />
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot with readings', () => {
      const readings = [
        createReading(45.2, 'Quiet', 1, 'Tonal Sound'),
        createReading(68.7, 'Normal', 3, 'Voice / Music'),
        createReading(82.1, 'Noisy', 5, 'Traffic / Heavy Machinery'),
      ];

      const { toJSON } = render(
        <NoiseHistory readings={readings} />
      );

      expect(toJSON()).toMatchSnapshot();
    });

    it('should match snapshot without timestamps', () => {
      const readings = [
        createReading(60, 'Normal'),
      ];

      const { toJSON } = render(
        <NoiseHistory readings={readings} showTimestamps={false} />
      );

      expect(toJSON()).toMatchSnapshot();
    });
  });
});
