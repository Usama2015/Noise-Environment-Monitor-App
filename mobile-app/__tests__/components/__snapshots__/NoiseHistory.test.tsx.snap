// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoiseHistory Snapshot tests should match snapshot for empty state 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
      "paddingVertical": 40,
    }
  }
>
  <Text
    style={
      {
        "color": "#999",
        "fontSize": 16,
        "fontWeight": "600",
        "marginBottom": 4,
      }
    }
  >
    No readings yet
  </Text>
  <Text
    style={
      {
        "color": "#BBB",
        "fontSize": 13,
      }
    }
  >
    Start monitoring to see history
  </Text>
</View>
`;

exports[`NoiseHistory Snapshot tests should match snapshot with readings 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "600",
        "marginBottom": 12,
        "paddingHorizontal": 16,
      }
    }
  >
    Recent Readings
  </Text>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingHorizontal": 16,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      {
        "flex": 1,
        "maxHeight": 300,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "marginRight": 12,
              }
            }
          >
            ðŸ”´
          </Text>
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "marginRight": 8,
                  }
                }
              >
                82.1
                 dB
              </Text>
              <Text
                style={
                  {
                    "color": "#666",
                    "fontSize": 14,
                  }
                }
              >
                Noisy
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#999",
                  "fontSize": 11,
                  "fontStyle": "italic",
                  "marginTop": 2,
                }
              }
            >
              Traffic / Heavy Machinery
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#999",
              "fontSize": 11,
              "marginLeft": 8,
            }
          }
        >
          5m ago
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "marginRight": 12,
              }
            }
          >
            ðŸŸ¡
          </Text>
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "marginRight": 8,
                  }
                }
              >
                68.7
                 dB
              </Text>
              <Text
                style={
                  {
                    "color": "#666",
                    "fontSize": 14,
                  }
                }
              >
                Normal
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#999",
                  "fontSize": 11,
                  "fontStyle": "italic",
                  "marginTop": 2,
                }
              }
            >
              Voice / Music
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#999",
              "fontSize": 11,
              "marginLeft": 8,
            }
          }
        >
          3m ago
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "marginRight": 12,
              }
            }
          >
            ðŸŸ¢
          </Text>
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "marginRight": 8,
                  }
                }
              >
                45.2
                 dB
              </Text>
              <Text
                style={
                  {
                    "color": "#666",
                    "fontSize": 14,
                  }
                }
              >
                Quiet
              </Text>
            </View>
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#999",
                  "fontSize": 11,
                  "fontStyle": "italic",
                  "marginTop": 2,
                }
              }
            >
              Tonal Sound
            </Text>
          </View>
        </View>
        <Text
          style={
            {
              "color": "#999",
              "fontSize": 11,
              "marginLeft": 8,
            }
          }
        >
          1m ago
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`NoiseHistory Snapshot tests should match snapshot without timestamps 1`] = `
<View
  style={
    {
      "flex": 1,
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 16,
        "fontWeight": "600",
        "marginBottom": 12,
        "paddingHorizontal": 16,
      }
    }
  >
    Recent Readings
  </Text>
  <RCTScrollView
    contentContainerStyle={
      {
        "paddingHorizontal": 16,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      {
        "flex": 1,
        "maxHeight": 300,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#FAFAFA",
            "borderRadius": 8,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 8,
            "paddingHorizontal": 12,
            "paddingVertical": 12,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 16,
                "marginRight": 12,
              }
            }
          >
            ðŸŸ¡
          </Text>
          <View
            style={
              {
                "flex": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#333",
                    "fontSize": 14,
                    "fontWeight": "600",
                    "marginRight": 8,
                  }
                }
              >
                60.0
                 dB
              </Text>
              <Text
                style={
                  {
                    "color": "#666",
                    "fontSize": 14,
                  }
                }
              >
                Normal
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
